x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -19,6 +37,23 @@ interface EditServiceTypeProps {
   isDisabled?: boolean;
 }
 
+// TODO move these constants to a common file in Serverless as we have in Stack
+const BETA_LABEL = i18n.translate('xpack.serverlessSearch.betaLabel', {
+  defaultMessage: 'Beta',
+});
+
+const TECH_PREVIEW_LABEL = i18n.translate('xpack.serverlessSearch.techPreviewLabel', {
+  defaultMessage: 'Tech preview',
+});
+
+interface CustomOptionData {
+  _append: React.ReactNode[];
+  _prepend: React.ReactNode;
```

Comment:
Do these need to be generic names?

can we do `_icon` for `_prepend` since that what we seem to be using it for. and `_badges` for `_append` ?